hotel_reservation_summary_base/wizard/hotel_reservation_summary.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2020 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+from datetime import timedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.fields import Date
+
+
+class RoomReservationSummary(models.AbstractModel):
+    _name = "hotel.room.reservation.summary"
+    _description = "Alternative Room reservation summary"
+    _form_view = "implementation_module.set_in_implementation_model"
+
+    name = fields.Char(compute="_compute_name")
+    date_from = fields.Date(string="Date From", required=True)
+    date_to = fields.Date(string="Date To", required=True)
+    room_summary = fields.Text(
+        string="Room Summary", compute="_compute_room_summary"
+    )
+
+    @api.model
+    def action_generate_summary(self):
+        raise NotImplementedError
+
+    @api.multi
+    def get_summary_action(self):
+        self.ensure_one()
+        view = self.env.ref(self._form_view)
```

Comment:
A little amazed that this will reference the `_form_view` of the inheriting module

Reply:
\o/ OOP \o/